### PR TITLE
Add testing workflow with GH actions

### PR DESCRIPTION
**Pull request by piraka9011**
This is a direct port of the Travis CI tests to GH actions.

Signed-off-by: Anas Abou Allaban <allabana@amazon.com>

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.5
+    - name: Install dependencies
+      run: |
+        sudo apt update && sudo apt install -y enchant
+        pip3 install -U setuptools==44.0.0
+        pip3 install -r requirements_devel.txt
+        pip3 install -U git+https://github.com/colcon/colcon-bundle
+        pip3 install -U git+https://github.com/ros-infrastructure/rosdep
```
Comment (thomas-moulard) [inline .github/workflows/test.yml]: That seems fragile - can't we use the release?

Comment (piraka9011) [reply .github/workflows/test.yml]: This was in the `.travis.yml`. I'm assuming they wanted it this way to catch any breakage from the upstream deps.

Comment (thomas-moulard) [reply .github/workflows/test.yml]: For colcon-bundle, this makes sense, `rosdep` is not maintained by us that's why it's more surprising. We should definitely install `colcon-bundle` from master. Maybe it'd be good to check why they need master from rosdep with Matt directly?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Test colcon-ros-bundle
+on:
+  pull_request:
+  schedule:
+    - cron: '43 17 * * *'
```
Comment (zmichaels11) [inline .github/workflows/test.yml]: Can you add a comment explaining what this means in English?

File: .github/workflows/test.yml (hunk 1 above)
Comment (piraka9011) [reply .github/workflows/test.yml]: > Well there’s two sets of tests. One that tests against mainline of all the deps and one that tests against the current pypi version. That’s so we can try to catch any bugs in mainline before they are released, specifically the colcon-* packages. - Matt

File: .github/workflows/test.yml (hunk 2 above)
Comment (piraka9011) [reply .github/workflows/test.yml]: Done

**Pull request merged**